endpoints/setuid.go

```diff
@@ -42,37 +37,44 @@ func NewSetUIDEndpoint(cfg config.HostCookie, syncers map[openrtb_ext.BidderName
 
 		defer pbsanalytics.LogSetUIDObject(&so)
 
-		pc := usersync.ParsePBSCookieFromRequest(r, &cfg)
+		pc := usersync.ParseCookieFromRequest(r, &cfg)
 		if !pc.AllowSyncs() {
 			w.WriteHeader(http.StatusUnauthorized)
-			metricsEngine.RecordUserIDSet(metrics.UserLabels{
-				Action: metrics.RequestActionOptOut,
-			})
+			metricsEngine.RecordSetUid(metrics.SetUidOptOut)
 			so.Status = http.StatusUnauthorized
 			return
 		}
 
 		query := r.URL.Query()
 
-		familyName, err := getFamilyName(query, validFamilyNameMap)
+		syncerKey, err := getSyncerKey(query, syncers)
 		if err != nil {
 			w.WriteHeader(http.StatusBadRequest)
 			w.Write([]byte(err.Error()))
-			metricsEngine.RecordUserIDSet(metrics.UserLabels{
-				Action: metrics.RequestActionErr,
-			})
+			metricsEngine.RecordSetUid(metrics.SetUidSyncerUnknown)
+			so.Status = http.StatusBadRequest
+			return
+		}
+		so.Bidder = syncerKey
+
+		responseFormat, err := getResponseFormat(query, syncers[syncerKey])
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
```

Comment:
It's not possible to have an invalid default sync type since it's validated on server startup and an error is treated as fatal. With this context, do you think it's worth returning an empty string without an error? I need to handle the default case in code, even though it's not reachable in practice.

Reply:
Discussed offline. I'll change `getResponseFormat` to return an empty string if it's not provided or if PBS doesn't know the default.